### PR TITLE
8290066: Remove KNL specific handling for new CPU target check in IR annotation

### PR DESCRIPTION
**Pull request by jatin-bhateja**
- Newly added annotations query the CPU feature using white box API which returns the list of features enabled during VM initialization.
- With JVM flag UseKNLSetting, during VM initialization AVX512 features not supported by KNL target are disabled, thus we do not need any special handling for KNL in newly introduced IR annotations (applyCPUFeature, applyCPUFeatureOr, applyCPUFeatureAnd).

Please review and share your feedback.

Best Regards,
Jatin

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8290066](https://bugs.openjdk.org/browse/JDK-8290066): Remove KNL specific handling for new CPU target check in IR annotation

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)
 * [Sandhya Viswanathan](https://openjdk.org/census#sviswanathan) (@sviswa7 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9452/head:pull/9452` \
`$ git checkout pull/9452`

Update a local copy of the PR: \
`$ git checkout pull/9452` \
`$ git pull https://git.openjdk.org/jdk pull/9452/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9452`

View PR using the GUI difftool: \
`$ git pr show -t 9452`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9452.diff">https://git.openjdk.org/jdk/pull/9452.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/compiler/lib/ir_framework/TestFramework.java
Comment (vnkozlov) [inline test/hotspot/jtreg/compiler/lib/ir_framework/TestFramework.java:138]: I don't think we should add these flags to whitelist in these changes - they can affect generated code.
New RFE is filed already to handle such flags: [8289801](https://bugs.openjdk.org/browse/JDK-8289801)

Comment (jatin-bhateja) [reply test/hotspot/jtreg/compiler/lib/ir_framework/TestFramework.java]: Done.

**Review by vnkozlov — COMMENTED** [review]
Good. I will start testing.

**Review by vnkozlov — APPROVED** [review]
Testing passed.

**Comment by jatin-bhateja** [comment]
Hi @kvn do we need second approval for this. Facing integration issue.

**Comment by vnkozlov** [comment]
> Hi @kvn do we need second approval for this. Facing integration issue.

It is known issue which is investigated.
Yes, you need second review.

**Comment by jatin-bhateja** [comment]
@chhagedorn , can you kindly review and approve.

**Comment by jatin-bhateja** [comment]
Hi @dean-long , @chhagedorn , can you kindly check and approve this.

**Review by sviswa7 — APPROVED** [review]
Looks good to me.

**Comment by jatin-bhateja** [comment]
Thanks @sviswa7 , @vnkozlov

**Pull request closed without merging**
